srml/session/src/lib.rs

```diff
@@ -332,33 +343,47 @@ decl_module! {
 }
 
 impl<T: Trait> Module<T> {
-	/// Move on to next session: register the new authority set.
+	/// Move on to next session. Register new validator set and session keys. Changes
+	/// to the validator set have a session of delay to take effect. This allows for
+	/// equivocation punishment after a fork.
 	pub fn rotate_session() {
-		// Increment current session index.
 		let session_index = CurrentIndex::get();
 
-		let mut changed = Changed::take();
+		let changed = QueuedChanged::take();
```

Comment:
```suggestion
		let changed = QueuedChanged::get();
```

Should be enough, because otherwise you will call 2 times into the host and that is not necessary.